### PR TITLE
Convert recipe examples to includes

### PR DESCRIPTION
**Pull request by waveform80**
Makes it much easier to test things - no copying'n'pasting just run the
examples straight from the dir (after wiring stuff up)

### CODE REVIEW

File: docs/recipes.rst [hunk 1]
```diff
@@ -930,54 +385,24 @@ Measure temperature with an ADC
...
-    led = RGBLED(2, 3, 4)
-    red_pot = MCP3008(0)
-    green_pot = MCP3008(1)
-    blue_pot = MCP3008(2)
-
-    led.source = zip(red_pot.values, green_pot.values, blue_pot.values)
-
-    pause()
+.. literalinclude:: examples/rgbled_pot_2.py
+    :emphasize-lines: 8
```
Comment (lurch) [inline docs/recipes.rst]: I think you emphasized the wrong line? http://gpiozero.readthedocs.io/en/latest/recipes.html#full-color-led-controlled-by-3-potentiometers

**Comment by lurch** [comment]
Nice! But perhaps filenames like `all_on_traffichat.py` would have been more meaningful than `all_on_2.py` ? ;-)

File: docs/recipes.rst (hunk 1 above)
Comment (waveform80) [inline docs/recipes.rst]: Doh - I did - unfortunately the offline build (which I normally give a quick eye-ball before pushing a PR) doesn't include the emphasis for some reason. Must figure out a way I can test this better.

**Comment by waveform80** [comment]
Yeah, they would be - but my creativity only stretches so far at 11pm :) I'll have to leave it for now - but it ought to be fixed up at some point

File: docs/recipes.rst (hunk 1 above)
Comment (lurch) [inline docs/recipes.rst]: Could it be as simple as the offline build using the wrong theme or something?

Comment (waveform80) [inline docs/recipes.rst]: Offline build's using the default theme - unfortunately I think it's unaware of the emphasis styles. But then the offline build has to remain compatible with whatever downlevel version of sphinx is supported in Raspbian at the moment (debuild ensures that venv's aren't used during package build so you can try and use a later sphinx, but it won't work - you'll wind up on the raspbian installed version)

Comment (lurch) [inline docs/recipes.rst]: Ahhh, so maybe there should be a `doc_with_rtd_styling:` target in the Makefile? ;-)  (for testing purposes)

Comment (waveform80) [inline docs/recipes.rst]: Heh - maybe - don't think I'll have time for that any-time soon though!

**Pull request merged**